src/main/java/org/oscarehr/ws/rest/DocumentService.java

```diff
@@ -76,4 +89,77 @@ public Response saveDocumentToDemographic(DocumentTo1 documentT) {
+        LoggedInInfo loggedInInfo = getLoggedInInfo();
+        
+        // Validate access
+        if (!securityInfoManager.hasPrivilege(loggedInInfo, "_edoc", SecurityInfoManager.WRITE, "")) {
+            logger.error("Write Access Denied _edoc for provider {}", loggedInInfo.getLoggedInProviderNo());
+            return Response.status(Response.Status.FORBIDDEN).entity(createResponseMap(documentTo1.getFileName(), "Failed", "Access Denied")).build();
+        }
+
+        // Validate input
+        if (isInvalidDocument(documentTo1)) { return Response.status(Response.Status.BAD_REQUEST).entity(createResponseMap(documentTo1.getFileName(), "Failed", "Missing required fields: fileName, fileContent, contentType, or queue")).build(); }
+        if (documentTo1.getQueue() < 1) { return Response.status(Response.Status.BAD_REQUEST).entity(createResponseMap(documentTo1.getFileName(), "Failed", "Invalid queue: must be 1 or greater")).build(); }
+
+        if (StringUtils.isEmpty(documentTo1.getSource())) { documentTo1.setSource("REST API"); }
+
+        DocumentConverter documentConverter = new DocumentConverter();
+        Document document = documentConverter.getAsDomainObject(loggedInInfo, documentTo1);
+        
+        // Set program ID if available
+        ProgramProvider programProvider = programManager2.getCurrentProgramInDomain(loggedInInfo, loggedInInfo.getLoggedInProviderNo());
+        if (programProvider != null && programProvider.getProgramId() != null) { document.setProgramId(programProvider.getProgramId().intValue()); }
+
+        try {
+            document = documentManager.createDocument(loggedInInfo, document, documentTo1.getDemographicNo(), documentTo1.getProviderNo(), documentTo1.getFileContents());
+        } catch (IOException e) {
+            logger.error("Document could not be saved: {}", documentTo1.getFileName(), e);
+            return Response.status(Response.Status.INTERNAL_SERVER_ERROR).entity(createResponseMap(documentTo1.getFileName(), "Failed", "Internal error: " + e.getMessage())).build();
+        }
+
+        Integer queueId = documentManager.addDocumentToQueue(loggedInInfo, document.getDocumentNo(), documentTo1.getQueue());
+
+        documentTo1 = documentConverter.getAsTransferObject(loggedInInfo, document);
+        if (queueId != null) { documentTo1.setQueue(queueId); }
+
+        return Response.ok(documentTo1).build();
+    }
+
+    private boolean isInvalidDocument(DocumentTo1 doc) {
+        if (doc == null || 
+            doc.getFileName() == null || doc.getFileName().isEmpty() || 
+            doc.getFileContents() == null || doc.getFileContents().length == 0 || 
+            doc.getContentType() == null || doc.getContentType().isEmpty() ||
+            doc.getQueue() == null) {
+            return true;
+        }
+    
+        if (StringUtils.isEmpty(doc.getProviderNo()) || !doc.getProviderNo().matches("\\d+")) {
```

Comment:
Provider numbers are alpha numeric, stored as varchar(6) in the database, and treated as Strings.  matches("\\d+") may be a limitation in the future.